src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs

```diff
@@ -368,6 +593,116 @@ End Sub
                End Class");
         }
 
+        [Fact]
+        public async Task ArgumentException_GenericParameterName_DoesNotWarn()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                public class Class
+                {
+                    public void Test<TEnum>(TEnum first)
+                    {
+                        throw new System.ArgumentException(""first is incorrect"", nameof(TEnum));
```

Comment:
Shouldn't we also have a test that validates that `throw new System.ArgumentException(""TEnum"");` is fixed to `throw new System.ArgumentException(null, nameof(TEnum));`?